doc/index.md

```diff
@@ -125,6 +125,33 @@ vector with it:
     // vi == {1,2,2,3,3,3,4,4,4,4,5,5,5,5,5,...}
 ~~~~~~~
 
+### View constness
+
+View is a range, pair of iterators. In order to work, and work fast, many views need to cache some data.
+In order to keep iterators small sized - this cached data stored in the view itself, and iterators holds only pointers to their view.
+Because of cache, many views have non-const only `begin()`/`end()` (though they may return const-iterators).
+Const version of `begin()`/`end()` (where provided) - are truly const, hence - tread-safe.
+But since they can't cache - const versions usually slower then non-const.
+
+Constness of view is not constness of iterators. Non-const view may hold const iterators.
+
+Use non-const views whenever possible. If you need thread-safety - work with view copies in threads, don't share.
+
+### View validity
+
+View is guaranteed to be valid, only until it adopted range remains constant all over its span.
+Iterator/sentinel validity alone is not enough.
+
+~~~~~~~{.cpp}
+    using namespace ranges;
+    std::forward_list<int> list = {1,2,3,4};
+    auto v = list | view::drop_last(1);
+    iter_swap(advance(list.begin(), 1), advance(list.begin(), 2));
```

Comment:
This code is invalid. You probably meant `next`, not `advance`. And I don't think this code actually invalidates the view `v`. Before the `iter_swap`, `v` refers to the range [1,2,3] and afterward to [1,3,2]. Or have I misunderstood?

Reply:
> You probably meant next, not advance

Right.

> And I don't think this code actually invalidates the view v ...

You're right, I think, I imagined myself that on `iter_swap` iterators will be invalidated. 

I'll try change example.

Reply:
You know, from the other side, it looks good without example. I remove it.